File: in_Java/0034 Find First and Last Position of Element in Sorted Array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] res = new int[2];
        res[0] = find1(nums,target);
        res[1] = find2(nums,target);
        return res;
    }
    private int find1(int[] nums, int t){
        if(nums == null || nums.length == 0) return -1;
        int i = 0, j = nums.length-1;
        while(i + 1 < j){
            int m = i + (j-i)/2;
            if(nums[m] == t) j = m;
            else if(nums[m] > t) j = m-1;
            else i = m + 1;
        }
        if(nums[i] == t) return i;
        if(nums[j] == t) return j;
        return -1;
    }
    private int find2(int[] nums, int t){
        if(nums == null || nums.length == 0) return -1;
        int i = 0, j = nums.length-1;
        while(i + 1 < j){
            int m = i + (j-i)/2;
            if(nums[m] == t) i = m;
            else if(nums[m] > t) j = m-1;
            else i = m + 1;
        }
        if(nums[j] == t) return j;
        if(nums[i] == t) return i;
        return -1;
    }
}

File: in_Python/0034 Find First and Last Position of Element in Sorted Array.py
class Solution(object):
    def searchRange(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        i = 0
        j = len(nums) - 1
        res = []
        res.append(-1)
        res.append(-1)
        if nums == None or len(nums) == 0:
            return res
        while i < j:
            m = i + (j-i)/2
            if nums[m] < target:
                i = m + 1
            else:
                j = m
        if nums[i] == target:
            res[0] = i
        i = 0
        j = len(nums) - 1
        while i+1 < j:
            m = i + (j-i)/2
            if nums[m] > target:
                j = m - 1
            elif nums[m] == target:
                i = m
            else:
                i = m + 1
        if nums[j] == target:
            res[1] = j
        elif nums[i] == target:
            res[1] = i           
        return res